pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -89,38 +130,117 @@ function make_PyTerminal(app: PyScriptApp) {
 
-            if (this.isAuto()) {
-                this.classList.add('py-terminal-hidden');
-                this.autoShowOnNextLine = true;
-            } else {
-                this.autoShowOnNextLine = false;
-            }
+            this.setupPosition(app);
+        }
 
+        // implementation of the Stdio interface
+        stdout_writeline(msg: string) {
+            this.outElem.innerText += msg + '\n';
             if (this.isDocked()) {
-                this.classList.add('py-terminal-docked');
+                this.scrollTop = this.scrollHeight;
+            }
+            if (this.autoShowOnNextLine) {
+                this.classList.remove('py-terminal-hidden');
+                this.autoShowOnNextLine = false;
             }
+        }
 
-            logger.info('Registering stdio listener');
-            app.registerStdioListener(this);
+        stderr_writeline(msg: string) {
+            this.stdout_writeline(msg);
         }
+        // end of the Stdio interface
+    }
 
-        isAuto() {
-            return this.hasAttribute('auto');
+    return PyTerminalPre;
+}
+
+declare const Terminal: typeof TerminalType;
+
+function make_PyTerminal_xterm(app: PyScriptApp) {
+    /** The <py-terminal> custom element, which automatically register a stdio
+     *  listener to capture and display stdout/stderr
+     */
+    class PyTerminalXterm extends PyTerminalBaseClass {
+        outElem: HTMLDivElement;
+        moduleResolved: boolean;
+        xtermReadyPromise: Promise<TerminalType>;
+        xterm: TerminalType;
+        cachedStdOut: Array<string>;
+        cachedStdErr: Array<string>;
+
+        constructor() {
+            super();
+            this.moduleResolved = false;
+            this.cachedStdOut = [];
+            this.cachedStdErr = [];
+
+            //Required to make xterm appear properly
+            this.style.width = '100%';
+            this.style.height = '100%';
         }
 
-        isDocked() {
-            return this.hasAttribute('docked');
+        async connectedCallback() {
+            this.outElem = document.createElement('div');
+            //this.outElem.className = 'py-terminal';
+            this.appendChild(this.outElem);
+
+            this.setupPosition(app);
+
+            this.xtermReadyPromise = this.setupXterm();
+            await this.xtermReadyPromise;
+        }
+
+        async setupXterm() {
+            // eslint-disable-next-line
+            // @ts-ignore
+            if (globalThis.Terminal == undefined) {
+                //xterm module proper
+                //eslint-disable-next-line
+                //@ts-ignore
+                await import('https://cdn.jsdelivr.net/npm/xterm@5.1.0/lib/xterm.js');
+
+                const cssTag = document.createElement('link');
+                cssTag.type = 'text/css';
+                cssTag.rel = 'stylesheet';
+                cssTag.href = 'https://cdn.jsdelivr.net/npm/xterm@5.1.0/css/xterm.css';
```

Comment:
see previous question ... shouldn't we at least put this CDN base URL somewhere else so we can patch one base URL and reuse it to bootstrap via CDN, if we want to go with the CDN approach?

Reply:
Agreed, I will pull it out to `_xterm_cdn_base_url`, and we can continue the convo about the CDN approach as well.